lib/network.sh

```diff
@@ -46,7 +46,6 @@ if [[ "${BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY}" == "true" ]]; then
   export BOOT_MODE="${BOOT_MODE:-UEFI}"
   export BARE_METAL_PROVISIONER_NETWORK="${BARE_METAL_PROVISIONER_NETWORK:-fd2e:6f44:5dd8:b856::/64}"
 else
-  export BOOT_MODE="${BOOT_MODE:-legacy}"
```

Comment:
Nobody is providing default value now, if `else` happens.

The following `if`  on L53 will fail trying to read undefined value.

Reply:
I have modified this by the time the comment arrived but thanks anyways!